presto-native-execution/presto_cpp/main/PrestoServer.h

```diff
@@ -83,9 +83,16 @@ class PrestoServer {
     return nodeState_;
   }
 
-  void setNodeState(NodeState nodeState) {
-    nodeState_ = nodeState;
-  }
+  /// Set worker into the SHUTTING_DOWN state even if we aren't shutting down.
+  /// This will prevent coordinator from sending new tasks to this worker.
+  void detachWorker();
+
+  /// Set worker into the ACTIVE state if we aren't shutting down.
+  /// This will enable coordinator to send new tasks to this worker.
+  void maybeAttachWorker();
+
+  /// Changes this node's state.
+  void setNodeState(NodeState nodeState);
```

Comment:
How is this different from detachWorker and maybeAttachWorker?

Reply:
Technically, detach wouldn't detach if we are already detached.
But the end result would be 'detached'.

Attach, on the other hand, wouldn't attach if we are detached, because we are shutting down.

That's the difference.